File: java-api-core/src/com/kinvey/java/KinveyDownloadFileException.java
package com.kinvey.java;


import com.kinvey.java.model.FileMetaData;

public class KinveyDownloadFileException extends KinveyException{

    private FileMetaData downloadedFileMetaData;

    public KinveyDownloadFileException(String reason, String fix, String explanation) {
        super(reason, fix, explanation);
    }

    public KinveyDownloadFileException(String reason) {
        super(reason);
    }

    public FileMetaData getDownloadedFileMetaData() {
        return downloadedFileMetaData;
    }

    public void setDownloadedFileMetaData(FileMetaData downloadedFileMetaData) {
        this.downloadedFileMetaData = downloadedFileMetaData;
    }
}

File: java-api-core/src/com/kinvey/java/KinveyDownloadFileException.kt
package com.kinvey.java


import com.kinvey.java.model.FileMetaData

class KinveyDownloadFileException : KinveyException {

    var downloadedFileMetaData: FileMetaData? = null

    constructor(reason: String, fix: String, explanation: String) : super(reason, fix, explanation)

    constructor(reason: String) : super(reason)
}
